### PR TITLE
 WaitForPodBySelector can deal with deployment rollouts 

### PR DESCRIPTION
**Pull request by manno**

### CODE REVIEW

File: internal/kubernetes/cluster.go [hunk 1]
```diff
@@ -327,34 +266,55 @@ func (c *Cluster) WaitUntilPodBySelectorExist(ctx context.Context, namespace, se
...
 
 // WaitForPodBySelector combines WaitUntilPodBySelectorExist and
 // WaitForPodBySelectorRunning, as one doesn't check if the pod is ready and
 // the other would fail immediately if the pod does not exist.
 func (c *Cluster) WaitForPodBySelector(ctx context.Context, namespace, selector string, timeout time.Duration) error {
+	// We might hit the deployment in the middle of a rollout. We can't know if this an old pod or a new pod.
 	if err := c.WaitUntilPodBySelectorExist(ctx, namespace, selector, timeout); err != nil {
 		return err
 	}
+	// But now some time has passed and we can make sure all pods are running
```
Comment (jimmykarily) [inline internal/kubernetes/cluster.go:317]:
```suggestion
	// Make sure all pods are running
```
`some time has passed` makes it sound as if we rely on random delays which is not true. No matter how fast things happen, at this point we know that some pods do exist. If it's a fresh install, these are the new pods. If it's an upgrade, these may be a mix of old + new pods. There is no way they can be just old pods, at least not with the manifests we use. In that case, we just wait until they balance out.

File: internal/kubernetes/cluster.go [hunk 2]
```diff
@@ -231,14 +189,6 @@ func (c *Cluster) WaitForPodRunning(ctx context.Context, namespace, podName stri
 	return wait.PollImmediate(time.Second, timeout, c.IsPodRunning(ctx, podName, namespace))
 }
 
-func (c *Cluster) WaitForJobCompleted(ctx context.Context, namespace, jobName string, timeout time.Duration) error {
```
Comment (jimmykarily) [inline internal/kubernetes/cluster.go:234]:
```~/workspace/epinio/installer (pod-ready-detection)$ go build -o installer cmd/epinio-installer/main.go 
# github.com/epinio/installer/internal/installer
internal/installer/component_actions.go:42:23: ca.cluster.WaitForJobCompleted undefined (type *"github.com/epinio/installer/internal/kubernetes".Cluster has no field or method WaitForJobCompleted)
```
We should have some CI that at least compiles and lints the code

File: internal/kubernetes/cluster.go (hunk 1 above)
Comment (manno) [reply internal/kubernetes/cluster.go:317]: I see. Actually, I wonder if the func still makes sense. All the comments are off.
Looks like `WaitForPodBySelectorRunning` can be the new `WaitForPodBySelector`, as it now checks for existence, too. It won't fail anymore if no pod exists.

However, I'm not sure that will cover all the cases. Having the other check in front of our new one, did indeed make sure that "some time has passed", e.g. the cluster was at least once in a state where pods for that selector existed.

Comment (jimmykarily) [reply internal/kubernetes/cluster.go:317]: on a fresh cluster we want to:
- wait until some pod for that selector appears
- wait until all pods for that selector are ready (hopefully nothing will delete them)

incidentally, the same plan above will work for upgrades too

**Pull request merged**